### ISSUE TITLE
Help wanted, how do I install the 3mf plugin

### ISSUE THREAD

**Issue by mitchngwen** [f3d-app/f3d#1567]
Thank you very much for this program it is very handy!

I got  f3d working on my debian based system and the files render with a 3d model but I would like to get 3mf files to do the same thing. I see that there are plugins available, but I have no idea how to install them. There is one that I saw on here that add 3mf support. Could someone walk me through on how to install the plugin and get the plugin it working?

**Comment by Meakk** [comment]
How did you install F3D on Debian? (Repository, .deb file, manual install?)
Can you copy the output of `f3d --version`?

**Comment by mitchngwen** [comment]
I did the manual install.

`$ f3d --version
F3D 2.5.0

F3D - A fast and minimalist 3D viewer
Version: 2.5.0.
Build date: 2024-07-01 20:59:05.
Build system: Linux 64-bits.
Compiler: GNU 8.4.0.
External rendering module: ON.
Raytracing module: OFF.
VTK version: 9.3.0-3457-gecf3646d64 (date: 20240620).
Copyright (C) 2019-2021 Kitware SAS.
Copyright (C) 2021-2024 Michael Migliore, Mathieu Westphal.
License BSD-3-Clause.
By Michael Migliore, Mathieu Westphal and Joachim Pouderoux.
`

**Comment by Meakk** [comment]
Ok then you should have 3MF support.  
Can you confirm the command `f3d --readers-list | grep 3mf` returns something?

**Comment by mitchngwen** [comment]
`~$ f3d --readers-list | grep 3mf
3MF        assimp  3D Manufacturing Format                 3mf   model/3mf       `

With 3mf in red letters.

**Comment by Meakk** [comment]
Then you have 3mf support, the plugin is already loaded.
What happens when you try to open the file?

**Comment by mitchngwen** [comment]
![Screenshot_20240818_102121](https://github.com/user-attachments/assets/f557de12-8dec-41ab-902c-5bb1d0079198)

**Comment by mwestphal** [comment]
So it works, you are just missing the tumbnail, is that correct ?

**Comment by mwestphal** [comment]
What do you get if you run:

`mimetype path/to/file.3mf` ?

**Comment by mitchngwen** [comment]
> So it works, you are just missing the tumbnail, is that correct ? 
Correct
![Screenshot_20240818_133756](https://github.com/user-attachments/assets/c3c0921f-e729-410b-9591-867a93df859b)

**Comment by Meakk** [comment]
I think it's related to a crash issue at exit in the assimp library that has been fixed since then.  
@mwestphal do you confirm the fix is not included in F3D 2.5?

**Comment by Meakk** [comment]
Related: https://github.com/f3d-app/f3d/issues/1301

@mitchngwen can you try the nightly build?

**Comment by mitchngwen** [comment]
Sure, I'll give it a try and let you know.

**Comment by mwestphal** [comment]
Not related to #1301 , @mitchngwen is missing the mime type for some reason . I'll need to check which package provides mimetypes in debian

**Comment by mitchngwen** [comment]
![Screenshot_20240818_155611](https://github.com/user-attachments/assets/10f7f803-edd6-47aa-a389-6c07208e0116)

No difference.

**Comment by mwestphal** [comment]
Can you share the failing file @mitchngwen ?

**Comment by mwestphal** [comment]
I think the package you need is `media-types`. `apt install media-types`.

**Comment by mitchngwen** [comment]
I can not install it.
```
Package media-types is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
```
[Gridfinity 2x4x3 Container.zip](https://github.com/user-attachments/files/16685834/Gridfinity.2x4x3.Container.zip)

All 3mf files are like this.

**Comment by mwestphal** [comment]
You need to find out how to install mimetypes. Please share your exact distribution if you want me to find the name of the package.

**Comment by mwestphal** [comment]
TLDR: `apt install shared-mime-info`

Longer version:

Using debian12 docker image, here is what I get:

```
apt update
apt install libfile-mimeinfo-perl wget unzip
wget https://github.com/user-attachments/files/16685834/Gridfinity.2x4x3.Container.zip
unzip Gridfinity.2x4x3.Container.zip
mimetype mimetype Gridfinity\ 2x4x3\ Container/Gridfinity\ 2x4x3\ Container.3mf
```

```
Gridfinity 2x4x3 Container/Gridfinity 2x4x3 Container.3mf: model/3mf
```

If I check `/usr/share/mime`  I find that there is actually two definition of mime types for 3mf:
`packages/freedesktop.org.xml` and ` model/3mf.xml`.

These have been installed by the packages:
`shared-mime-info`

But this package should be installed already.

**Comment by mwestphal** [comment]
@Meakk maybe we should add that info in the DESKTOP_INTEGRATION.md ?

**Comment by mitchngwen** [comment]
> I think the package you need is `media-types`. `apt install media-types`.

I can not install media types.

![Screenshot_20240821_191459](https://github.com/user-attachments/assets/a24dc3ed-2082-4442-b43f-14ebb56fbde9)

**Comment by mitchngwen** [comment]
Feren OS 2023.04

I seem to be missing ` model/3mf.xml`

![Screenshot_20240821_191927](https://github.com/user-attachments/assets/815c9bb9-cd01-4a09-9ddc-01de05d5222a)
![Screenshot_20240821_192223](https://github.com/user-attachments/assets/11b3127a-3553-4970-a662-65a2fd2a8677)

**Comment by mitchngwen** [comment]
This has been fixed. I had to manually add 3mf mime type. What a pain. Some weirdness with my system I guess. Thank you for all your help, guys.

**Comment by mwestphal** [comment]
It is possible the 3mf was added quite recently in `freedesktop.org.xml` and is not present in your version. We could add a `f3d-3mf-formats.xml` but we are trying to avoid duplicating mime types with the one provided by the system.

**Comment by mitchngwen** [comment]
Ah, perhaps make it available in case someone else runs into this issue? Thank you guys again for being so helpful.
